csharp/src/Google.Protobuf/ParseContext.cs

```diff
@@ -65,6 +65,29 @@ internal static void Initialize(ref ReadOnlySpan<byte> buffer, ref ParserInterna
             ctx.state = state;
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal static void Initialize(ref ReadOnlySpan<byte> input, out ParseContext ctx)
+        {
+            Initialize(ref input, DefaultRecursionLimit, out ctx);
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal static void Initialize(ref ReadOnlySpan<byte> input, int recursionLimit, out ParseContext ctx)
+        {
```

Comment:
Does input need to be ref here?

I think comments explaining how and why these Initialize overloads are different from the existing overload that takes a ReadOnlySpan on line 62.

`internal static void Initialize(ref ReadOnlySpan<byte> buffer, ref ParserInternalState state, out ParseContext ctx)`

Reply:
As above, I removed passing by `ref` and added some comments with my best understanding of how they were being used!